keps/sig-node/20190423-expose-tty-probes.md

```diff
@@ -0,0 +1,123 @@
+---
+title: KEP Template
+authors:
+  - "@umohnani8"
+owning-sig: sig-api-machinery
+participating-sigs:
+  - sig-architecture
+reviewers:
+  - TBD
+  - "@yujuhong"
+approvers:
+  - "@Random-Liu"
+  - "@yujuhong"
+editor: TBD
+creation-date: 2019-04-15
+last-updated: 2019-04-15
+status: implemented
+see-also:
+replaces:
+superseded-by:
+---
+
+# Expose tty option for probes
+
+## Table of Contents
+
+A table of contents is helpful for quickly jumping to sections of a KEP and for highlighting any additional information provided beyond the standard KEP template.
+[Tools for generating][] a table of contents from markdown are available.
+
+- [Title](#title)
+  - [Table of Contents](#table-of-contents)
+  - [Release Signoff Checklist](#release-signoff-checklist)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+  - [Proposal](#proposal)
+    - [Risks and Mitigations](#risks-and-mitigations)
+  - [Design Details](#design-details)
+    - [Test Plan](#test-plan)
+    - [Graduation Criteria](#graduation-criteria)
+  - [Implementation History](#implementation-history)
+  - [Alternatives [optional]](#alternatives-optional)
+
+[Tools for generating]: https://github.com/ekalinin/github-markdown-toc
+
+## Release Signoff Checklist
+For enhancements that make changes to code or processes/procedures in core Kubernetes i.e., [kubernetes/kubernetes], we require the following Release Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These checklist items _must_ be updated for the enhancement to be released.
+
+- [ ] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+## Summary
+
+Some readiness and liveness probes have exec commands that use `/bin/sh -i -c`, especially when deploying postgres and mysql. However, these probes fail with an `ioctl error` that happens because tty is set to false by default. Setting `-i` in thses exec command helps to pick the default binary paths and libraries from the start up scripts without having to specify the full paths in the command. Adding an optional boolean tty field to the probes gives the users the flexibility of turning n tty whenever using `/bin/sh -i -c` in their exec commands.
```

Comment:
Can you give an actual example that fails?

Reply:
Added example of failure.